tests/index/in_memory/test_persist_data.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import pytest
+
+from docarray import BaseDoc, DocList
+from docarray.index import InMemoryExactNNIndex
+from docarray.typing import NdArray
+
+pytestmark = [pytest.mark.slow, pytest.mark.index]
+
+
+class SimpleDoc(BaseDoc):
+    simple_tens: NdArray[10]
+    simple_text: str
+
+
+class ListDoc(BaseDoc):
+    docs: DocList[SimpleDoc]
+    simple_doc: SimpleDoc
+    list_tens: NdArray[20]
+
+
+class MyDoc(BaseDoc):
+    docs: DocList[SimpleDoc]
+    list_docs: DocList[ListDoc]
+    my_tens: NdArray[30]
+
+
+def test_subindex_index(tmp_path):
```

Comment:
I think we should also check that find and find_subindex work

Reply:
Sure, I'll add that